### PR TITLE
Support additional slave arguments

### LINKED ISSUE #34: Pass the visible flag to Python
FMI defines a instantiation option visible.
For completeness, we should pass it to the Python slave.

### PR DESCRIPTION
**Pull request by markaren**
~A bit ugly solution to #34. I was not able to use **kwargs from the C-API, so I use a dict object. This solution looks uglier in the Python API, but thats not the main use-case so..~

Now uses `*args, **kwargs` approach.

Closes #34

### CODE REVIEW

**Comment by markaren** [comment]
I think the alternative to this is the make the `Fmi2Slave` ctor look like this

```py
def __init__(self, *args, *kwargs):
```

where `*args` would hold `instance_name` and additional arguments goes into `**kwargs`. I think the issue with the C API is to call a ctro with only `**kwargs`.

**Comment by markaren** [comment]
Can confirm that the above works. Question is then whats more preferable.

**Comment by markaren** [comment]
I'm happy with this.

**Comment by fcollonval** [comment]
Nice work @markaren 

Before noticing your PR, I proposed an alternative approach https://github.com/NTNU-IHB/PythonFMU/issues/34#issuecomment-585216592

What do you think?

**Comment by markaren** [comment]
The loggingOn flag is intentionally left on the C++ side. We may not have to expose it to the Pyhton code and let the native side handle it. But I dunno. It's easy to add if needed.

File: pythonfmu/fmi2slave.py [hunk 1]
```diff
@@ -31,9 +31,14 @@ class Fmi2Slave(ABC):
...
 
-    def __init__(self, instance_name: str):
+    def __init__(self, *args, **kwargs):
         self.vars = OrderedDict()
-        self.instance_name = instance_name
+        self.instance_name = args[0]
+
+        self.visible = False
+        if "visible" in kwargs:
+            self.visible = kwargs["visible"]
```
Comment (fcollonval) [inline pythonfmu/fmi2slave.py]: You could simplify this in python:
```python
self.visible = kwargs.get("visible", False)
```

File: pythonfmu/pythonfmu-export/cpp/PyObjectWrapper.cpp [hunk 2]
```diff
@@ -62,9 +62,12 @@ PyObjectWrapper::PyObjectWrapper(const std::string& instanceName, const std::str
     if (pClass == nullptr) {
         handle_py_exception("[ctor] PyObject_GetAttr");
     }
-    PyObject *argList = Py_BuildValue("s", instanceName.c_str());
-    pInstance_ = PyObject_CallFunctionObjArgs(pClass, argList, nullptr);
-    Py_DECREF(argList);
+
+    PyObject* args = Py_BuildValue("s", instanceName.c_str());
+    PyObject* kwargs = Py_BuildValue("{s:i}", "visible", visible);
```
Comment (fcollonval) [inline pythonfmu/pythonfmu-export/cpp/PyObjectWrapper.cpp]: Nice to know that a dictionary does it for kwargs.

```suggestion
PyObject* kwargs = Py_BuildValue("{sssi}", "instance_name", instanceName.c_str(), "visible", visible);
```

`:` should not be present

File: pythonfmu/tests/test_fmi2slave.py [hunk 3]
```diff
@@ -15,17 +15,25 @@
 
 @pytest.mark.parametrize("model", ["theModelName", None])
 def test_Fmi2Slave_constructor(model):
+
     class Slave(Fmi2Slave):
+
         modelName = model
+
+        def __init__(self, instance_name):
+            super().__init__(instance_name)
```
Comment (fcollonval) [inline pythonfmu/tests/test_fmi2slave.py]: This is unneeded if you just pass the arguments to super.

Comment (markaren) [reply pythonfmu/tests/test_fmi2slave.py]: ` super().__init__(instance_name)` you mean?

**Comment by markaren** [comment]
> I would put all arguments in kwargs to reduce the risk of errors when writing a customized daughter class and to provide some context on the argument (namely the keys).

I tried that, but the C API won't let me. Which is why I also ended up with args.

File: pythonfmu/tests/test_fmi2slave.py (hunk 3 above)
Comment (markaren) [reply pythonfmu/tests/test_fmi2slave.py]: Ahh, I get it now

File: pythonfmu/fmi2slave.py (hunk 1 above)
Comment (markaren) [reply pythonfmu/fmi2slave.py]: Nice

**Comment by fcollonval** [comment]
I tried a all-keyword-arg approach in e866f012f94db288419c2bcc1f8e40149b152216. If you think this is ok we can merge it. Otherwise feel free to reset your PR to your latest commit and let's merge this one 😃

**Comment by markaren** [comment]
Looks good

**Pull request merged**